script.rb

```diff
@@ -0,0 +1,281 @@
+## by Hillarioh
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+
+    enum = to_enum
+
+    while i < size
+      yield(enum.next)
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    i = 0
+
+    enum = to_enum
+
+    while i < size
+      yield(enum.next, i)
+
+      i += 1
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    arrayed = []
+
+    my_each { |i| arrayed << i if yield(i) == true }
+
+    arrayed
+  end
+
+  def my_all?(val = nil)
+    return true if size.zero?
+
+    arrayed = []
+
+    if val.class == Regexp
+      my_each { |i| arrayed << val.match?(i) }
+    elsif block_given?
+      my_each { |i| arrayed << yield(i) }
+
+    elsif !val.nil?
+      my_each do |i|
+        arrayed << if i.class == val.class
+                     (i == val)
+
+                   elsif val.class == Class
+                     if i.is_a? val
+                       true
+                     else
+                       false
+                     end
+                   else
+                     false
+                   end
+      end
+
+    elsif !val
+      my_each do |i|
+        arrayed << false if i.nil? || i == false
+      end
+    end
+
+    state = true
+    j = 0
+
+    while j < arrayed.size
+      if arrayed[j] != true
+        state = false
+        break
+      end
+      j += 1
+    end
+
+    state
+  end
+
+  def my_any?(val = nil)
+    return false if size.zero?
+
+    arrayed = []
+
+    if val.class == Regexp
+      my_each { |i| arrayed << val.match?(i) }
+    elsif block_given?
+      my_each { |i| arrayed << yield(i) }
+
+    elsif !val.nil?
+      my_each do |i|
+        arrayed << if i.class == val.class
+                     (i == val)
+
+                   elsif val.class == Class
+                     i.is_a? val ? true : false
+                   else
+                     false
+                   end
+      end
+
+    elsif !val
+      my_each do |i|
+        arrayed << (i == false || i.nil? ? false : true)
+      end
+    end
+
+    state = false
+
+    j = 0
+
+    while j < arrayed.size
+      if arrayed[j] == true
+        state = true
+        break
+      end
+      j += 1
+    end
+
+    state
+  end
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_any?(Integer) == array.any?(Integer) # false
  ```